lib/builtin.pm

```diff
@@ -414,15 +418,12 @@ A complete list is in L<perlrecharclass/Whitespace>.
 
 C<trim> is equivalent to:
 
-    $str =~ s/\A\s+|\s+\z//urg;
+    s/\A\s+//u, s/\s+\z//u for my $trimmed = $str;
```

Comment:
Maybe `my $trimmed = $str =~ s/\A\s+//ur =~ s/\s+\z//ur;`

Reply:
I considered this first as it looks clearer but it would be two new strings instead of one. Maybe irrelevant compared to the regex performance though.

Reply:
I was curious (and didn't expect such difference) in benchmark: https://gist.github.com/happy-barney/b3cfa0017569fe6666eda68c389de62d

(maybe this benchmark (or similar) along with `builtin::trim` can be part of codebase so people can learn ?